### PR TITLE
Generate nice doc html for all libs

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/50510

### CODE REVIEW

**Comment by asdil12** [comment]
The doc should appear under https://os-autoinst.github.io/os-autoinst-distri-opensuse/ after this pr got merged.

File: script/generateLibDocs.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    git fetch origin gh-pages:gh-pages
+
+    echo "Documentation of libs" > docs/index.html
+    echo "<ul><li>libs/</li><ul>" >> docs/index.html
+
+    for docfile in $(cd lib ; grep -s ^=head * | grep .pm | cut -d. -f1 | sort -u) ; do
+        echo "Generating docs for lib/${docfile}.pm"
+        echo "<li><a href="${docfile}.html">${docfile}.pm</a></li>" >> docs/index.html
+        touch docs/${docfile}.html
```
Comment (SergioAtSUSE) [inline script/generateLibDocs.sh]: touch? Why don't you let pod2html create the file?

File: script/generateLibDocs.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        pod2html --infile=lib/${docfile}.pm --outfile=docs/${docfile}.html
+        # remove line that contains perl version and breaks diff
+        sed -i '/^<link rev="made" href="mailto:/d' docs/${docfile}.html
+        sed -i 's|^</head>|<link rel="stylesheet" href="style.css" />\n</head>|' docs/${docfile}.html
+        sed -i "s|></title>|>lib/${docfile}.pm</title>|" docs/${docfile}.html
+        sed -i "s|^</ul>|</ul><h1>lib/${docfile}.pm</h1>|" docs/${docfile}.html
```
Comment (SergioAtSUSE) [inline script/generateLibDocs.sh]: After which list are you appending this h1?

Comment (foursixnine) [inline script/generateLibDocs.sh]: Looks a bit rudimentary but we can look at it if needed, later on, it's bash afterall...

**Comment by asdil12** [comment]
@foursixnine See http://artemis.suse.de:8000/

**Pull request merged**